### ISSUE TITLE
New provider digital ocean

### ISSUE THREAD

**Issue by MitkoTschimev** [Dev-Thought/nx-plugins#52]
**Is your feature request related to a problem? Please describe.**
I want to use digital ocean as provider.

**Describe the solution you'd like**
Integrate digital ocean as provider for nx-deploy-it

**Check which provider is affected:**
[] Azure
[] AWS
[] Google Cloud Platform

**Comment by MitkoTschimev** [comment]
@iangregsondev
Can you provide us some information:
-  how to deploy a static app to digital ocean
    - can option would be nice too if it exists
- how to set up a server with a docker image

**Comment by iangregsondev** [comment]
Hi @MitkoTschimev , I don't know if I am going to be of much help here :-) But I will try.

My situation is a little different - well I presume many are in this situation. I actually run docker on digital ocean and I connect to it using docker-machine and I can deploy my services that way.

BUT ... if there is an API available here 

https://developers.digitalocean.com/guides/

I just find it easier to manage everything with docker.

Maybe somebody else can give some insight also?

I know that digital ocean provides exceptional service and at much cheaper prices compared to azure and was. In fact, that is one of the reasons I use them :-)

But as I say I run docker there so I only connect via docker-machine and I can deploy anything I want

**Comment by MitkoTschimev** [comment]
Serving static files via a docker instance is a way to expensive because you pay for a web server 24/7 and I think digital ocean has a static file server solution too. But let's figure it out ;)

**Comment by iangregsondev** [comment]
@MitkoTschimev Yep, I totally agree as I say my situation is a little different as I serve a couple of microservices and some static sites from the same machine - I pay around 5 dollars a month and I also have digital ocean spaces (s3 storage).

I actually did start off with azure but it was just working out a little too expensive. 

Actually something worth looking at, although I don't know if they have it yet - but something like Azure functions / AWS lambdas - serverless functions - they only come alive when they are actually doing some work - per second billing etc.

This is 100% a good use case!

I did start off working with azure functions and its something that I would want to look at in the future - may be digital ocean has something similar.

**Comment by MitkoTschimev** [comment]
If you want to get started with azure functions try the nestjs / expressjs integration with nx-deploy-it.
